inspirehep/celery.py

```diff
@@ -25,15 +25,27 @@
 from __future__ import absolute_import, division, print_function
 
 import logging
+import os
 
-from flask_celeryext import create_celery_app
+from flask_celeryext import AppContextTask, create_celery_app
+from sqlalchemy.exc import InvalidRequestError, OperationalError
 
-from .factory import create_app
+from inspirehep.factory import create_app
 
 
-celery = create_celery_app(
-    create_app(LOGGING_SENTRY_CELERY=True)
-)
+LOGGER = logging.getLogger(__name__)
+
+
+class CeleryTask(AppContextTask):
+
+    def on_failure(self, exc, task_id, args, kwargs, einfo):
+        if isinstance(exc, InvalidRequestError) or isinstance(exc, OperationalError):
```

Comment:
* maybe also add `psycopg2.OperationalError`, as that's the one we get sometimes? e.g. https://sentry.inspirehep.net/inspire-prod/hep/issues/5982/?query=is%3Aunresolved%20operationalerror
* You can combine this as
```suggestion
        if isinstance(exc, (InvalidRequestError, OperationalError)):
```